### DISCUSSION TITLE
이벤트 캡처링, 버블링

### DISCUSSION THREAD

**Discussion by JUDONGHYEOK** [woowacourse-study/2022-woowahan-bansanghwe#5] [JavaScript]
이벤트 캡처링과 버블링은 무엇일까요? 또 각각 언제 사용하면 좋을까요?

**Comment by JUDONGHYEOK** [comment]
https://stackoverflow.com/questions/24786769/real-world-example-where-event-capturing-is-necessary-preferred

**Comment by greenblues1190** [comment]
https://www.quirksmode.org/js/events_order.html 정리

#### 이벤트 순서의 두 가지 모델

넷스케이프와 마이크로소프트는 서로 다른 결론에 도달했습니다.
- Netscape는 부모 요소의 이벤트가 먼저 발생한다고 말했습니다. 이를 이벤트 캡처링이라고 합니다.
- Microsoft는 자식 요소의 이벤트가 우선 적용된다고 유지했습니다. 이것을 이벤트 버블링이라고 합니다.

두 이벤트 명령은 근본적으로 반대입니다. Explorer는 이벤트 버블링만 지원합니다. Mozilla, Opera 7 및 Konqueror는 둘 다 지원합니다. 구형 Opera와 iCab은 둘 다 지원하지 않습니다.

#### W3C 모델

W3C는 이 투쟁에서 중간 입장을 취하기로 매우 현명하게 결정했습니다. [W3C 이벤트 모델](http://www.w3.org/TR/2000/REC-DOM-Level-2-Events-20001113/) 에서 발생하는 모든 이벤트는 대상 요소에 도달할 때까지 먼저 캡처된 다음 다시 버블링됩니다.

#### 요약

브라우저마다 달랐는데 두가지 모두 남음
